include/chopper/pack/hierarchical_binning.hpp

```diff
@@ -214,20 +223,23 @@ class hierarchical_binning
         // initialize first row
         if (estimate_union)
         {
+            size_t sum = 0;
```

Comment:
Can you move this definition outside the if/else (you do this line in both branches).